claims_p1.go

```diff
@@ -172,108 +172,49 @@ func (c *P1Claims) SetVSI(v string) error {
 
 // Codecs
 
-func (c *P1Claims) FromCBOR(buf []byte) error {
-	err := c.FromUnvalidatedCBOR(buf)
-	if err != nil {
-		return err
-	}
-
-	err = c.Validate()
-	if err != nil {
-		return fmt.Errorf("validation of PSA claims failed: %w", err)
-	}
+// this type alias is used to prevent infinite recursion during marshaling.
+type p1Claims P1Claims
 
-	return nil
-}
-
-func (c *P1Claims) FromUnvalidatedCBOR(buf []byte) error {
+func (c *P1Claims) UnmarshalCBOR(buf []byte) error {
 	c.Profile = nil // clear profile to make sure we take it from buf
 
-	err := dm.Unmarshal(buf, c)
+	a := (*p1Claims)(c) // prevent the decoder invoking this method again
+	err := dm.Unmarshal(buf, a)
```

Comment:
It's probably just a question of taste, but I'd go with merging these two lines (like you did in a similar situation in the ccatoken PR).